### PR TITLE
feat(): Added slice health metrics

### PR DESCRIPTION
**Pull request by gourishkb**
This PR adds the slice health related metrics.

### CODE REVIEW

File: tests/hub/hub_suite_test.go [hunk 1]
```diff
@@ -148,6 +148,7 @@ var _ = BeforeSuite(func() {
 		testSliceEventRecorder,
 		mf,
 	)
+	sr.ReconcileInterval = 5 * time.Second
```
Comment (rajendra-avesha) [inline tests/hub/hub_suite_test.go:151]: Can you please add IT for the same start the UT for the same

**Review by rajendra-avesha — CHANGES_REQUESTED** [review]

File: pkg/hub/controllers/slice_controller.go [hunk 2]
```diff
@@ -62,6 +62,8 @@ func NewSliceReconciler(mc client.Client, er *events.EventRecorder, mf metrics.M
 		counterSliceCreationFailed: mf.NewCounter("slice_creation_failed_total", "Slice creation failed in worker", []string{"slice"}),
 		counterSliceUpdationFailed: mf.NewCounter("slice_updation_failed_total", "Slice updation failed in worker", []string{"slice"}),
 		counterSliceDeletionFailed: mf.NewCounter("slice_deletion_failed_total", "Slice deletion failed in worker", []string{"slice"}),
+		gaugeSliceUp:               mf.NewGauge("slice_up", "Kubeslice slice health status", []string{}),
+		gaugeComponentUp:           mf.NewGauge("slice_component_up", "Kubeslice slice component health status", []string{"component"}),
```
Comment (kmjayadeep) [inline pkg/hub/controllers/slice_controller.go:66]:
```suggestion
		gaugeComponentUp:           mf.NewGauge("slice_component_up", "Kubeslice slice component health status", []string{"kubeslice_slice_component"}),
```

File: tests/hub/hub_suite_test.go (hunk 1 above)
Comment (gourishkb) [reply tests/hub/hub_suite_test.go:151]: Added, please review again

File: pkg/hub/controllers/slice_controller.go (hunk 2 above)
Comment (gourishkb) [reply pkg/hub/controllers/slice_controller.go]: Done

File: tests/utils/metrics.go [hunk 3]
```diff
@@ -1,38 +1,61 @@
...
 }
 
-func GetGaugeMetricFromRegistry(r *prometheus.Registry, name string) []float64 {
-	mf, _ := r.Gather()
-	values := []float64{}
-	for _, m := range mf {
-		if *m.Name == name {
-			for _, v := range m.Metric {
-				values = append(values, v.Gauge.GetValue())
+func matchLabels(lp []*dto.LabelPair, ml map[string]string) bool {
```
[the remark this answers is not in the document]
Comment (gourishkb) [reply tests/utils/metrics.go:42]: https://github.com/kubeslice/worker-operator/pull/236/commits/578ced71e60bd4a0877d2f4d92b640e33a2700f7 the commit is part of the PR already, not sure why this diff is still seen.

File: tests/hub/slice_controller_test.go [hunk 4]
```diff
@@ -347,7 +347,15 @@ var _ = Describe("Hub SliceController", func() {
 
 			for _, c := range cs {
 				Expect(string(c.ComponentHealthStatus)).Should(Equal("Error"))
+				m, err := utils.GetGaugeMetricFromRegistry(MetricRegistry, "kubeslice_slice_component_up", map[string]string{
+					"slice_component": c.Component,
```
Comment (kmjayadeep) [inline tests/hub/slice_controller_test.go:351]: please add slice label as well

File: tests/spoke/service_import_test.go [hunk 5]
```diff
@@ -151,6 +153,37 @@ var _ = Describe("ServiceImportController", func() {
 				}
 				return true
 			}, time.Second*30, time.Millisecond*250).Should(BeTrue())
+
```
Comment (kmjayadeep) [inline tests/spoke/service_import_test.go:156]: the diff is confusing. can you close and reopen the pr to make sure the diff is correct?

**Review by kmjayadeep — CHANGES_REQUESTED** [review]

**Comment by gourishkb** [comment]
Closing this PR for https://github.com/kubeslice/worker-operator/pull/242

**Pull request closed without merging**